packages/core/types/ts3.6/index.d.ts

```diff
@@ -166,7 +166,7 @@ export interface SagaMiddleware<C extends object = {}> extends Middleware {
    * @param saga a Generator function
    * @param args arguments to be provided to `saga`
    */
-  run<S extends Saga>(saga: S, ...args: Parameters<S>): Task
+  run<Args extends any[], TReturn>(saga: Saga<Args, TReturn>, ...args: Args): Task<TReturn>
```

Comment:
This is a change in the declared signature. What does it give us? Can you come up with any real-life scenario? 

I know *some* stuff about the difference between the two but I'd like to hear out your thoughts and motivation.

Reply:
It is a change, but given it's just changing inferred generic type parameters, I don't think it's a breaking change. I don't think it's common for people to specify the saga's type, they just pass the saga.

I'm not sure what kind of description or specific use cases you're looking for beyond the example I gave in the PR description, and of course the tests showing that the task's type was not being inferred before.

> In my application I have a function runSaga that takes a saga as a parameter and returns the Task that the saga returns when it's begun. I want to get the proper type of the Task's result, but am not able to do so with the current Saga type definition.

Reply:
Please let me know if I can be clearer or if you think the tests don't demonstrate the improvement — getting an specific, inferred type for the task's return instead of `any`

Reply:
But does it provide a meaningful improvement anywhere? Can we add a test that would "prove" that?

Reply:
I'm confused, are you not seeing the two tests that are a part of this PR?

Reply:
I see them but I don't see from them how exactly changing the generics here has made them better. 

What I mean is that to make them work it would probably be enough to just extract the return type like this:
```ts
type SagaReturnType<S extends Saga<any, any>> = S extends Saga<any, infer R> ? R : never

// ...
run<S extends Saga>(saga: S, ...args: Parameters<S>): Task<SagaReturnType<S>>
```

Reply:
Ah I see — you understand the point and value of adding the return type to `Saga`, but you have a way you would prefer to implement this piece of it.

I don't have a preference between the two forms, the way I wrote it was just how I'd figured out how to do it.

Went ahead and changed it to your suggested form 👍